rig-core/src/transcription.rs

```diff
@@ -0,0 +1,243 @@
+    pub data: Vec<u8>,
+    /// The file name to be used in the request
+    pub filename: String,
+    /// The language used in the response from the transcription model provider
+    pub language: String,
+    /// The prompt to be sent to the transcription model provider
+    pub prompt: Option<String>,
+    /// The temperature sent to the transcription model provider
+    pub temperature: Option<f64>,
+    /// Additional parameters to be sent to the transcription model provider
+    pub additional_params: Option<serde_json::Value>,
+}
+
+/// Builder struct for a transcription request
+///
+/// Example usage:
+/// ```rust
+/// use rig::{
+///     providers::openai::{Client, self},
+///     transcription::TranscriptionRequestBuilder,
+/// };
+///
+/// let openai = Client::new("your-openai-api-key");
+/// let model = openai.transcription_model(openai::WHISPER_1).build();
+///
+/// // Create the completion request and execute it separately
+/// let request = TranscriptionRequestBuilder::new(model, "~/audio.mp3".to_string())
+///     .temperature(0.5)
+///     .build();
+///
+/// let response = model.transcription(request)
+///     .await
+///     .expect("Failed to get transcription response");
+/// ```
+///
+/// Alternatively, you can execute the transcription request directly from the builder:
+/// ```rust
+/// use rig::{
+///     providers::openai::{Client, self},
+///     transcription::TranscriptionRequestBuilder,
+/// };
+///
+/// let openai = Client::new("your-openai-api-key");
+/// let model = openai.transcription_model(openai::WHISPER_1).build();
+///
+/// // Create the completion request and execute it directly
+/// let response = TranscriptionRequestBuilder::new(model, "~/audio.mp3".to_string())
+///     .temperature(0.5)
+///     .send()
+///     .await
+///     .expect("Failed to get transcription response");
+/// ```
+///
+/// Note: It is usually unnecessary to create a completion request builder directly.
+/// Instead, use the [TranscriptionModel::transcription_request] method.
+pub struct TranscriptionRequestBuilder<M: TranscriptionModel> {
+    model: M,
+    data: Vec<u8>,
+    filename: String,
+    language: String,
+    prompt: Option<String>,
+    temperature: Option<f64>,
+    additional_params: Option<serde_json::Value>,
+}
+
+impl<M: TranscriptionModel> TranscriptionRequestBuilder<M> {
+    pub fn new(model: M) -> Self {
+        TranscriptionRequestBuilder {
+            model,
+            data: vec![],
+            filename: "".to_string(),
+            language: "en".to_string(),
+            prompt: None,
+            temperature: None,
+            additional_params: None,
+        }
+    }
+
+    /// Sets the data for the request
+    pub fn data(mut self, filename: &str, data: Vec<u8>) -> Self {
+        self.filename = filename.to_string();
+        self.data = data;
+        self
+    }
+
+    /// Load the specified file into data
+    pub fn load_file(self, path: &str) -> Self {
```

Comment:
```suggestion
    pub fn load_file<P>(self, path: P) -> Self
    where
        P: AsRef<path::Path> {
```
This allows you to pass anything that *can* be a path which is flexible + it's actually generating a function per type of argument you provide. We could apply this technique to other places in the codebase, but I haven't fully explored the pros and cons (higher binary sizes code-readability vs flexibility and speed).

Many things in `std` do it this way!